pkgs/servers/monitoring/uptime-kuma/default.nix

```
@@ -0,0 +1,56 @@
+{ pkgs, lib, stdenv, fetchFromGitHub, fetchzip, substituteAll, nixosTests, iputils }:
+let
+  deps = import ./composition.nix { inherit pkgs; };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "uptime-kuma";
+  version = "1.18.0";
+
+  src = fetchFromGitHub {
+    owner = "louislam";
+    repo = "uptime-kuma";
+    rev = finalAttrs.version;
+    sha256 = "sha256-XcHgwYswkaqKrRWxE2TsLEZHQN535boPMhaLWxV3pn8=";
+  };
+
+  patches = [
+    # Fixes the permissions of the database being not set correctly
+    # See https://github.com/louislam/uptime-kuma/pull/2119
+    ./fix-database-permissions.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace server/ping-lite.js \
+      --replace "/bin/ping" "${iputils}/bin/ping" \
+    --replace "/sbin/ping" "${iputils}/bin/ping" \
+    --replace "/sbin/ping6" "${iputils}/bin/ping"
+  '';
+
+  installPhase = ''
+    mkdir -p $out/
+    cp -r . $out/
+    ln -s ${deps.package}/lib/node_modules/uptime-kuma/node_modules/ $out
+    ln -s ${finalAttrs.passthru.ui} $out/dist
+  '';
+
+  passthru = {
+    tests = {
+      inherit (nixosTests) uptime-kuma;
+    };
```

```suggestion
    tests.uptime-kuma = nixosTests.uptime-kuma;
```

I think this is significantly cleaner and if the name changes it should be updated everywhere so it's not like they could go out of sync

but this is mostly preference